Mathlib/RingTheory/Bialgebra/Equiv.lean

```diff
@@ -0,0 +1,275 @@
+/-
+Copyright (c) 2024 Amelia Livingston. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Amelia Livingston
+-/
+import Mathlib.RingTheory.Coalgebra.Equiv
+import Mathlib.RingTheory.Bialgebra.Hom
+
+/-!
+# Isomorphisms of `R`-bialgebras
+
+This file defines bundled isomorphisms of `R`-bialgebras. We simply mimic the early parts of
+`Mathlib/Algebra/Algebra/Equiv.lean`.
+
+## Main definitions
+
+* `BialgEquiv R A B`: the type of `R`-bialgebra isomorphisms between `A` and `B`.
+
+## Notations
+
+* `A ≃ₐc[R] B` : `R`-bialgebra equivalence from `A` to `B`.
+-/
+
+set_option autoImplicit true
+
+open BigOperators
+
+universe u v w u₁ v₁
+
+variable {R : Type u} {A : Type v} {B : Type w} {C : Type u₁}
+
+open Bialgebra
+
+/-- An equivalence of bialgebras is an invertible bialgebra homomorphism. -/
+structure BialgEquiv (R : Type u) [CommSemiring R] (A : Type v) (B : Type w)
+    [Semiring A] [Semiring B] [Algebra R A] [Algebra R B]
+    [CoalgebraStruct R A] [CoalgebraStruct R B] extends A ≃ₗc[R] B, A ≃* B where
+
+attribute [nolint docBlame] BialgEquiv.toMulEquiv
+attribute [nolint docBlame] BialgEquiv.toCoalgEquiv
+
+@[inherit_doc BialgEquiv]
+notation:50 A " ≃ₐc[" R "] " B => BialgEquiv R A B
```

Comment:
The `c` stands for coalgebra, right?

Reply:
yeah, it's supposed to be like a combination of the `CoalgEquiv` and `AlgEquiv` notation. But it's hard to guess. Is there any other notation you'd suggest?

Reply:
Ah, I see, the point is that there are both the `a` and the `c`, so it is bialgebra hom. It makes sense.